### PR TITLE
Disable HSTS header in case of NOTLS=1

### PR DESCRIPTION
**Pull request by john721**
If kore is built with NOTLS, no need to force the browser to use https.

### CODE REVIEW

**Comment by jorisvink** [comment]
I think it's more sensible to remove all the hsts related things if NOTLS is set, not just ifdef the default value to 0.

.joris

**Comment by john721** [comment]
Please check the new changes.

File: src/http.c [hunk 1]
```diff
@@ -1424,12 +1424,14 @@ http_response_normal(struct http_request *req, struct connection *c,
 		}
 	}
 
+	#if !defined(KORE_NO_TLS)
```
Comment (jorisvink) [inline src/http.c]: Please don't indent preprocessor items.

**Comment by jorisvink** [comment]
So I leave a comment on your diff and you decide to close the PR all together? That's too bad, it was a sensible change.

**Comment by john721** [comment]
I was thinking the following scenario:
A user is hosting a HTTPS server on port 443. And he wants people who connects via http port 80 to use https port 443 instead. So he must handle connection from port 80. What can he do? I'm not sure how, but out of my mind is to run another kore instance on port 80 which is built with NOTLS=1 to do that.

But what in my pull request will stop the NOTLS=1 one from sending HSTS header. So I think that is not the correct way to do so. Maybe tuning the "http_hsts_enable" is a good option to do that:
- If built with NOTLS=1, by default it's 0, and use the config file to enable HSTS header.
- If built with NOTLS=0, since the successful connections are https already, there is no need to send HSTS header, so either leave the default value as 0 or remove HSTS if NOTLS=0 can do the work.

Comments? I'm happy to modify what you think is good for kore.

**Comment by jorisvink** [comment]
Or we change nothing and on the binary built without TLS you make sure http_hsts_enable is set to 0 in the configuration file.

That's probably the best thing to do rather then my earlier idea.

**Pull request closed without merging**